core/tests/test_utils.py

```diff
@@ -80,6 +81,68 @@ def empty_environ():
         os.environ['HTTP_HOST'], os.environ['SERVER_PORT'])
 
 
+def get_filepath_from_filename(filename, rootdir):
```

Comment:
What if (possibly in the future) two files have the same filename? Maybe detect that case and throw exception? Or have separate backend test that verifies that all files have different filenames (probably better since that means you can reference that test from here, and break early in the for loop here)

Reply:
I can add a backend test but I have two doubts:

- Is it necessary for files to have distinct names always? The test may fail if we later have any case which requires two same filenames.
- The test will import the `test_utils` file so I cannot it call it back in `test_utils` since that will create a circular dependency.

Reply:
Probably a separate backend test would be better then, due to the circular dependency. And I think you can assume different filenames for now, and if they need to be the same, we can handle that later. (Better to add a check now than to ignore that case and have silent failures or unexpected behaviour later.)

Reply:
I am sorry I don't clearly understand what you mean by separate backend test. Do we need to add a test which does not have class inheritance from `test_utils`?

Reply:
It should be a regular backend test, but I don't see why you'd need to call it from test_utils.

Reply:
Yes, it is fine if I do not call it but you suggested earlier:

```
probably better since that means you can reference that test from here, and break early in the for loop here
```

So, should I write part to check for same names in some common utility file?

Reply:
Oh -- "reference" in the original note just means "leave a comment".

You can write this in base_test.py (but add clear documentation to explain the context).

Reply:
Done.